tests/unit/package_managers/yarn/test_main.py

```diff
@@ -238,3 +261,88 @@ def test_set_yarnrc_configuration(mock_write: mock.Mock) -> None:
 def test_generate_environment_variables(yarn_env_variables: list[EnvironmentVariable]) -> None:
     result = _generate_environment_variables()
     assert result == yarn_env_variables
+
+
+@pytest.mark.parametrize(
+    "yarn_input_packages, package_components",
+    (
+        (
+            [{"type": "yarn", "path": "."}],
```

Comment:
> "yarn_input_packages,

You lost me here to be fair. It took me a while to realize this is how you perform a fixture override (I had to scroll through the docs). The most confusing part to me was that you actually return a dummy value from the `yarn_input_packages` which is actually never used because it would be always overriden, right?

I'm also missing some test IDs here so I can at least identify what the intended purpose of each test case is.

Do you think [indirect parametrization](https://docs.pytest.org/en/7.1.x/example/parametrize.html#apply-indirect-on-particular-arguments) would help here a tiny bit [1]? IOW

```
diff --git a/tests/unit/package_managers/yarn/test_main.py b/tests/unit/package_managers/yarn/test_main.py
index ca31b86c..3bba2705 100644
--- a/tests/unit/package_managers/yarn/test_main.py
+++ b/tests/unit/package_managers/yarn/test_main.py
@@ -25,8 +25,8 @@ from cachi2.core.rooted_path import RootedPath
 
 
 @pytest.fixture
-def yarn_input_packages() -> list[dict[str, str]]:
-    return [{"type": "yarn"}]
+def yarn_input_packages(request: pytest.FixtureRequest) -> list[dict[str, str]]:
+    return request.param
 
 
 @pytest.fixture
@@ -308,6 +308,7 @@ def test_generate_environment_variables(yarn_env_variables: list[EnvironmentVari
             ],
         ),
     ),
+    indirect=["yarn_input_packages"]
 )
 @mock.patch("cachi2.core.package_managers.yarn.main._resolve_yarn_project")
 @mock.patch("cachi2.core.package_managers.yarn.project.Project.from_source_dir")
```

I'd be okay if you decide to disagree with the above, since I'd propose some test cleanup at some point anyway, see my reasoning below.

[1] Our parametrize lists are massive in general, so I think long term I think we should extract these lists to constants and keep the parametrize decorator init compact (I know we have a ton of precedent for the way you're proposing). I think would this would ultimately help overall readability of the test cases because you can still know which test a given parametrize relates to which is IMO a problem currently as parametrize tends to span multiple pages.

Reply:
The above works for me :+1: 

I had never used or seen the indirect parametrization before, so TIL :smile: